dateparser/date.py

```diff
@@ -122,9 +122,15 @@ def sanitize_date(date_string):
 
 
 def get_date_from_timestamp(date_string, settings):
-    if RE_SEARCH_TIMESTAMP.search(date_string):
-        date_obj = datetime.fromtimestamp(int(date_string[:10]))
-        date_obj = apply_timezone_from_settings(date_obj, settings)
+    match = RE_SEARCH_TIMESTAMP.search(date_string)
+    if match:
+        seconds = int(match.group(1))
+        millis = int(match.group(2) or 0)
+        micros = int(match.group(3) or 0)
+        date_obj = datetime.fromtimestamp(seconds)
+        date_obj = date_obj.replace(microsecond=millis * 1000 + micros)
+        if date_obj:
```

Comment:
I am puzzled by why this `if` statement is needed, although according to the code coverage report it is indeed visited. What happens if you remove it, which tests fail?